okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -786,6 +806,14 @@ open class OkHttpClient internal constructor(
       this.eventListenerFactory = eventListenerFactory
     }
 
+    // This lambda conversion is for Kotlin callers expecting a Java SAM (single-abstract-method).
+    @JvmName("-deprecated_eventListenerFactory")
```

Comment:
Also considered adding this:
```
    @Deprecated(
        message = "moved to object",
        replaceWith = ReplaceWith(
            expression = "" +
                "eventListenerFactory(object : EventListener.Factory {\n" +
                "  override fun create(call: Call) = block(call)\n" +
                "})",
            imports = [
              "okhttp3.Call", 
              "okhttp3.EventListener.Factory"
            ]),
        level = DeprecationLevel.ERROR)
    inline fun eventListenerFactory(
      crossinline block: (call: Call) -> EventListener
    )
```

Reply:
It works, but when you apply the advice it looks a bit weird ’cause it actually creates the lambda and then calls it.